### PR TITLE
Added details to Spark section

### PR DESCRIPTION
**Pull request by YasminMorshed**

### CODE REVIEW

File: modules/core/Spark.md [hunk 1]
```diff
@@ -1,38 +1,101 @@
...
+![PySpark logo](https://miro.medium.com/max/800/1*VNdaFCkls0gyJR0ddP1PCQ.png "PySpark logo")
 
-You could also set up a notebook on a databricks workspace or use [databricks connect](https://docs.databricks.com/dev-tools/databricks-connect.html).
+## What is PySpark?
 
-Please download flight-data from the [Spark-The-Definitive-Guide repo](https://github.com/databricks/Spark-The-Definitive-Guide/tree/master/data/flight-data) and put it somewhere that is accessible in your pyspark environemnt. 
+Apache Spark + Python = PySpark
 
-For examples and guidance on using pyspark please refer to this documentation [here](https://spark.apache.org/docs/latest/api/python/reference/api/pyspark.sql.SparkSession.html)
+Spark is a tool for managing and coordinating the execution of tasks on data across a cluster of computers. Apache Spark is written primarily in the Scala programming language. PySpark is an interface for Apache Spark in Python, which is used for real-time, large-scale data processing. It not only allows you to write Spark applications using Python APIs, but also provides the PySpark shell for interactively analyzing your data in a distributed environment.
```
Comment (tf75) [inline modules/core/Spark.md:9]: I would change this paragraph to this:
"Apache Spark is a multi-language engine for executing data engineering, data science, and machine learning on single-node machines or clusters. Apache Spark is written primarily in the Scala programming language. PySpark is an interface for Apache Spark in Python, which is used for real-time, large-scale data processing."

File: modules/core/Spark.md [hunk 2]
```diff
@@ -1,38 +1,101 @@
...
 
-1) To begin please read in the csv file '2015-summary.csv using the DataFrameReader with option 'inferSchema' and 'header' set to true
+## Exercise
+Take a look at [PySpark's installation instructions](https://spark.apache.org/docs/latest/api/python/getting_started/install.html "PySpark's Installation Instructions").
```
Comment (tf75) [inline modules/core/Spark.md:12]: This should link back to our installation instructions in setting up python environment

File: modules/core/Spark.md [hunk 3]
```diff
@@ -1,38 +1,101 @@
...
+To run this excercise you should be able to use your local Python environment with PySpark enabled.
 
-3) A key part part of pyspark is transformation of data. Return a new dataframe sorted by the column 'count' (the sort command will be useful here and .explain() which will showcase the dataframe lineage) Run .take() command again after the sort to see how this has changed what the top rows are. 
+You could also set up a notebook on a Databricks workspace or use [Databricks Connect](https://docs.databricks.com/dev-tools/databricks-connect.html).
 
-4) A partition in spark is an atomic chunk of data (logical division of data) stored on a node in the cluster. spark.sql.shuffle.partitions configures the number of partitions that are used when shuffling data for joins or aggregations. By default when we perform a shuffle Spark outputs 200 shuffle partitions, experiment with reducing these partitions with 'spark.conf.set'. Depending on the number it may impact runtime. 
+If you are stuck on setting up the project, refer to the instructions on [this](https://github.com/databricks/Spark-The-Definitive-Guide "this") page.
```
Comment (tf75) [inline modules/core/Spark.md:18]: The setting up python environment should be instruction enough to set this up? Have you gone through that page, do we need to add anything?

File: modules/core/Spark.md [hunk 4]
```diff
@@ -1,38 +1,99 @@
...
+
+    [Row(DEST_COUNTRY_NAME='United States', ORIGIN_COUNTRY_NAME='Romania', count=15), Row(DEST_COUNTRY_NAME='United States', ORIGIN_COUNTRY_NAME='Croatia', count=1), Row(DEST_COUNTRY_NAME='United States', ORIGIN_COUNTRY_NAME='Ireland', count=344), Row(DEST_COUNTRY_NAME='Egypt', ORIGIN_COUNTRY_NAME='United States', count=15)]
+
+3) A key part of PySpark is transformation of data. Return a new DataFrame sorted by the column 'count' (the `sort` command will be useful here and `.explain()` which will showcase the DataFrame lineage).
+
+```python
+flightData2015.sort("count").explain()
+```
+
+Adding the code above will show us the explain plan, so you should see something like this:
```
Comment (AdamProbert) [inline modules/core/Spark.md:53]: Might be worth linking to some documentation on what an explain plan is

File: modules/core/Spark.md [hunk 5]
```diff
@@ -1,38 +1,99 @@
...
+spark.conf.set("spark.sql.shuffle.partitions", "5")
+```
+
+5) Next, we are going to look at how to run SQL queries against DataFrame. There is no performance difference between writing SQL queries or writing DataFrame code. Create a temporary table using the `createOrReplaceTempView` function.
+
+```python
+flightData2015.createOrReplaceTempView("flight_data_2015")
+```
+
+6) Below is an example of querying using SQL against a dataframe:
```
Comment (AdamProbert) [inline modules/core/Spark.md:82]:
```suggestion
6) Below is an example of querying using SQL against a DataFrame:
```

File: modules/core/Spark.md [hunk 6]
```diff
@@ -1,38 +1,99 @@
...
+```sql
 sqlWay = spark.sql("""
 SELECT DEST_COUNTRY_NAME, count(1)
 FROM flight_data_2015
 GROUP BY DEST_COUNTRY_NAME
 """)
 ```
 
-how could you do this using just DataFrame code? Using sqlWay.explain() will also show how Spark will execute the query. 
+How could you do this using just DataFrame code? Using `sqlWay.explain()` will also show how Spark will execute the query.
```
Comment (AdamProbert) [inline modules/core/Spark.md:92]: Might be useful showing the equivalent without sql?

**Review by AdamProbert — CHANGES_REQUESTED** [review]

**Review by AdamProbert — APPROVED** [review]
Great additions! Nice one Yasmin 🚀

**Pull request merged**